src/newgrf_badge_config.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file newgrf_badge_config.h Functions related to NewGRF badge configuration. */
+
+#ifndef NEWGRF_BADGE_CONFIG_H
+#define NEWGRF_BADGE_CONFIG_H
+
+#include "newgrf.h"
+#include "newgrf_badge_type.h"
+
+class BadgeClassConfigItem {
+public:
+	std::string label; ///< Class label.
+	int column = -1; ///< UI column, feature-dependent.
+	int sort_order = -1; ///< Position within list, populated only by GetBadgeClassConfigItem.
+	bool show_icon = false; ///< Set if the badge icons should be displayed for this class.
```

Comment:
In `GUIBadgeClasses::Element` this is called `visible`.

Reply:
`show_icon` because `show_filter` will also be a thing.